src/DwitTech.AccountService.Core/Services/ActivationService.cs

```diff
@@ -11,40 +11,63 @@ namespace DwitTech.AccountService.Core.Services
 {
     public class ActivationService : IActivationService
     {
-        private readonly IConfiguration _configuration; //Config instance for GetBaseUrl method
+        private readonly IConfiguration _configuration;
         private readonly IUserRepository _userRepository;
+        private readonly IHttpClientFactory _httpClientFactory;
+
+        public ActivationService(IConfiguration configuration, 
+            IUserRepository userRepository, 
+            IHttpClientFactory httpClientFactory)
         private readonly IEmailService _emailService;
         public ActivationService(IConfiguration configuration, IEmailService emailService, IUserRepository userRepository)
         {
             _configuration = configuration;
             _userRepository = userRepository;
+            _httpClientFactory = httpClientFactory;
             _emailService = emailService;
         }
 
-        private string GetActivationCode()
+        private static string GetActivationCode()
         {
             var activationCode = StringUtil.GenerateUniqueCode();
             return activationCode;
         }
 
-        private string GetBaseUrl()
+        private async Task<string> GetActivationUrl(int userId)
         {
+            string baseUrl = _configuration["BASE_URL"];
             return _configuration.GetSection("BaseUrl").Value;
         }
 
         private string GetActivationUrl(int userId)
         {
             string baseUrl = GetBaseUrl();
```

Comment:
This line has been modified, since GetBaseUrl has been removed. The variable gets its value directly from config now. Please check master and fix as required.